funnel/views/venue.py

```diff
@@ -183,3 +189,25 @@ def update_venue_colors(profile, project, id, color):
             room.bgcolor = colors[room.scoped_name]
     db.session.commit()
     return jsonify(status=True)
+
+
+@app.route('/<profile>/<project>/venue/makeprimary', methods=['POST'])
+@funnelapp.route('/<project>/venue/makeprimary', methods=['POST'], subdomain='<profile>')
+@lastuser.requires_login
+@load_models(
+    (Profile, {'name': 'profile'}, 'g.profile'),
+    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
+    permission='edit-project')
+def venue_make_primary(profile, project):
+    form = VenuePrimaryForm()
+    if form.validate_on_submit():
+        venue = Venue.query.filter_by(suuid=form.venue.data, project=project).first_or_404()
+        if venue and venue == project.primary_venue:
```

Comment:
You don't need `if venue and` since `venue` is guaranteed to exist at this point. Defensive code makes it harder to find bugs later.